vscode_ext/src/extension.ts

```diff
@@ -363,7 +366,20 @@ export async function activate(context: ExtensionContext): Promise<void> {
 
 	// set doc root and needs.json file
 	client.onReady().then(async () => {
-		await read_settings(outChannel);
```

Comment:
Wouldn't it be better to put the sequel into the make_needs function (or even better) into the needs server?
Ideally it would work as a make and run the build if no file present or if it is outdated regarding the input. 
Note also that some auto_build === false in make_needs might prohibit file generation.